packages/clerk-js/src/ui/contexts/ClerkUIComponentsContext.tsx

```diff
@@ -241,6 +241,8 @@ export const useUserButtonContext = () => {
   const afterSwitchSessionUrl = ctx.afterSwitchSessionUrl || displayConfig.afterSwitchSessionUrl;
   const navigateAfterSwitchSession = () => navigate(afterSwitchSessionUrl);
 
+  const userProfileMode = !!ctx.userProfileUrl && !ctx.userProfileMode ? 'navigation' : ctx.userProfileMode;
```

Comment:
❓ why did we change this and the next ones?

Reply:
This is a necessary change to determine the `userProfileMode`, if the `ctx.userProfileUrl` is filled and no `ctx.userProfileMode` was provided the `userProfileMode` will be `navigation`